compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -1587,6 +1624,178 @@ unsafe extern "C" {
     ) -> &'a Value;
 }
 
+// FFI bindings for `DIBuilder` functions in the LLVM-C API.
+// Try to keep these in the same order as in `llvm/include/llvm-c/DebugInfo.h`.
+//
+// FIXME(#134001): Audit all `Option` parameters, especially in lists, to check
+// that they really are nullable on the C/C++ side. LLVM doesn't appear to
+// actually document which ones are nullable.
+unsafe extern "C" {
+    pub(crate) fn LLVMCreateDIBuilder<'ll>(M: &'ll Module) -> *mut DIBuilder<'ll>;
+    pub(crate) fn LLVMDisposeDIBuilder<'ll>(Builder: ptr::NonNull<DIBuilder<'ll>>);
+
+    pub(crate) fn LLVMDIBuilderFinalize<'ll>(Builder: &DIBuilder<'ll>);
+
+    pub(crate) fn LLVMDIBuilderCreateNameSpace<'ll>(
+        Builder: &DIBuilder<'ll>,
+        ParentScope: Option<&'ll Metadata>,
+        Name: *const c_uchar,
+        NameLen: size_t,
```

Comment:
Implementation note: I'm using `*const c_uchar` for pointer/length strings, even though the C-side type is `char*`, because the pointer types are compatible anyway, and using `c_uchar` (i.e. `u8`) on the Rust side means we don't need an explicit pointer cast when passing `str::as_ptr` or `<[u8]>::as_ptr`.

Reply:
Can you leave this as a comment for the future reader?

Reply:
Good suggestion, but since this applies to dozens of parameters (and hopefully more of `ffi.rs` in the future), I'll need to put it somewhere else, like in a module-level comment.